openmls/src/group/public_group/diff.rs

```diff
@@ -0,0 +1,240 @@
+use std::collections::HashSet;
+
+use openmls_traits::{crypto::OpenMlsCrypto, types::Ciphersuite, OpenMlsCryptoProvider};
+use serde::{Deserialize, Serialize};
+use tls_codec::Serialize as TlsSerialize;
+
+use crate::{
+    binary_tree::LeafNodeIndex,
+    error::LibraryError,
+    framing::{mls_auth_content::AuthenticatedContent, public_message::InterimTranscriptHashInput},
+    group::GroupContext,
+    messages::{proposals::AddProposal, ConfirmationTag, EncryptedGroupSecrets},
+    schedule::{psk::PreSharedKeyId, CommitSecret, JoinerSecret},
+    treesync::{
+        diff::{StagedTreeSyncDiff, TreeSyncDiff},
+        errors::ApplyUpdatePathError,
+        node::{
+            encryption_keys::EncryptionKeyPair, leaf_node::OpenMlsLeafNode,
+            parent_node::PlainUpdatePathNode, Node,
+        },
+        treekem::{DecryptPathParams, UpdatePath, UpdatePathNode},
+    },
+};
+
+use super::PublicGroup;
+
+pub(crate) mod apply_proposals;
+pub(crate) mod process_path;
+
+pub(crate) struct PublicGroupDiff<'a> {
+    original_group: &'a PublicGroup,
+    diff: TreeSyncDiff<'a>,
+    group_context: GroupContext,
+    interim_transcript_hash: Vec<u8>,
+    // Most recent confirmation tag. Kept here for verification purposes.
+    confirmation_tag: ConfirmationTag,
+}
+
+impl<'a> PublicGroupDiff<'a> {
+    /// Create a new [`PublicGroupDiff`] based on the given [`PublicGroup`].
+    pub(super) fn new(public_group: &'a PublicGroup) -> PublicGroupDiff<'a> {
+        Self {
+            original_group: public_group,
+            diff: public_group.treesync().empty_diff(),
+            group_context: public_group.group_context().clone(),
+            interim_transcript_hash: public_group.interim_transcript_hash.to_vec(),
```

Comment:
Can we use a getter here as well? It makes it easier to understand where it's used.